Added `contributor.first/lastName`

roll opened the pull request:
- fixes https://github.com/frictionlessdata/specs/issues/852

---

# Rationale

Please take a look at https://github.com/frictionlessdata/specs/issues/852. Personally, I'm not sure about this change but if we think about Data Package as just a language (or definition table) we create for humans/machines to communicate with each other than standardising two additional properties which obviously make sense is a logical move.

peterdesmet commented:
Thanks! To exemplify with a use case: we have introduced this properties for biodiversity datasets, to allow generating a **citation** as `Lastname F, Lastname F ...` (see https://github.com/gbif/ipt/issues/2087#issuecomment-1883249674). We currently have to work around `title` being required, but that is resolved in https://github.com/frictionlessdata/datapackage/pull/7 and is no longer required in frictionless v2.

Adopting this PR will as @roll writes "standardis[e] two additional properties", rather than communities inventing them.

nichtich commented:
If the use case is to generate citations that the data model should be aligned with citation data (or applications will need to reinvent citation generators). The most common citation generator and data model is [CSL](https://citationstyles.org/). It's data model for contributor names [is described here](https://citeproc-js.readthedocs.io/en/latest/csl-json/markup.html#name-fields) and internals as JSON Schema [here](https://github.com/citation-style-language/schema/blob/master/schemas/input/csl-data.json#L462). So in the best case there is support of name fields:

* `family` - surname minus any particles and suffixes
* `given` - given names, either full (“John Edward”) or initialized (“J. E.”)
* `suffix` - name suffix, e.g. “Jr.” in “John Smith Jr.” and “III” in “Bill Gates III”
* `non-dropping-particle` - name particles that are not dropped when only the surname is shown (“van” in the Dutch surname “van Gogh”) but which may be treated separately from the family name, e.g. for sorting
* `dropping-particle` - name particles that are dropped when only the surname is shown (“van” in “Ludwig van Beethoven”, which becomes “Beethoven”, or “von” in “Alexander von Humboldt”, which becomes “Humboldt”)

As simplicity is preferred and `suffix`, `non-dropping-particle` and `dropping-particle` can be detected automatically to some degree, I'd say it is enough to support `given` and `family` by these names. The `title` field should be kept, it maps to CSL name field `literal` to be used as [exemplified here](https://citeproc-js.readthedocs.io/en/latest/csl-json/markup.html#name-fields).

khusmann commented:
> If the use case is to generate citations that the data model should be aligned with citation data (or applications will need to reinvent citation generators). 

Well said, @nichtich , I strongly agree.

> @khusmann does this address your comment at frictionlessdata/specs#852 (comment)?

@peterdesmet yes, it does! I fully support this now.

roll commented:
Thanks @nichtich! It's been updated -- does it look OK now?

PietrH commented:
I prefer last name over family name, after all, not everyone has a family name at all. Other cultures have other practises, and where I can I have personally switched to asking for a single "username" in forms. In this case ,this isn't very practical, but I still believe we should avoid using Family over Last. 

https://softwareforgood.com/why-you-should-stop-asking-for-first-and-last-names-on-forms/

https://www.kalzumeus.com/2010/06/17/falsehoods-programmers-believe-about-names/

Names are really complicated!

peterdesmet commented on content/docs/specifications/data-package.md:
````diff
@@ -269,12 +269,16 @@ The people or organizations who contributed to this Data Package. It `MUST` be a
 ```
 
 - `title`: name/title of the contributor (name for person, name/title of organization)
+- `givenName`: a given names, either full (“John Edward”) or initialized (“J. E.”), if the contributor is a person
````
```suggestion
- `givenName`: given name(s), either full (“John Edward”) or initialized (“J. E.”), if the contributor is a person
```

peterdesmet commented on content/docs/specifications/data-package.md:
```diff
@@ -269,12 +269,16 @@ The people or organizations who contributed to this Data Package. It `MUST` be a
 
 - `title`: name/title of the contributor (name for person, name/title of organization)
+- `givenName`: a given names, either full (“John Edward”) or initialized (“J. E.”), if the contributor is a person
+- `familyName`: a surname minus any particles and suffixes if the contributor is a person
```
```suggestion
- `familyName`: familial name, if the contributor is a person
```

peterdesmet commented:
I personally don't know what the most inclusive way is to represent names, other than the single `name` which we already have (but that is problematic for citations).

- The [EML](https://lternet.edu/wp-content/uploads/2010/12/EMLHandbook-2.pdf) standard uses `givenName`, `surName`
- The currently referenced citeproc-js uses `given`, `family`
- Wikipedia uses [surname](https://en.wikipedia.org/wiki/Surname) (canonical for `Last name` or `Family name`) and [given name](https://en.wikipedia.org/wiki/Given_name)
- Bibtex uses a single [author](https://www.bibtex.com/f/author-field/) field, parsing is done on comma
- ...

I'm fine with an alternative where we don't add 2 new fields, but rather provide formatting rules for the existing `name` (cf. bibtex)

roll commented:
BTW I would take into account, that `datapackage.json` is still a machine-focused communication format so I would say that it's not that important how we exactly name properties if it provides desired functionality. So if it's `contributor.lastName` in JSON it doesn't mean that it must be `Last Name` in e.g. UI

nichtich commented:
> I'm fine with an alternative where we don't add 2 new fields, but rather provide formatting rules for the existing title

Parsing and formatting names is a form of art but not relevant to this specification. All we need to do is provide a way to store a common subset of name information. Existing standards suggest three fields (`given` and `family`=`surname` if available and applicable, `title` otherwise) is a good solution.

roll commented:
I've updated back to `first/lastName` but using more broad BibTeX definition. It now mention both first / given and last / family terms and exact name of the properties are not so important, imho. Also, added a recommendation to follow their guidance for `title`.

peterdesmet commented on content/docs/specifications/data-package.md:
````diff
@@ -268,13 +268,17 @@ The people or organizations who contributed to this Data Package. It `MUST` be a
 ```
 
-- `title`: name/title of the contributor (name for person, name/title of organization)
+- `title`: name or title if the contributor is an organization; name if the contributor is a person
````
```suggestion
- `title`: name of the contributor. It's `RECOMMENDED` to follow [BibTeX name format](https://www.bibtex.com/f/author-field/) when providing a person's name in `title`.
```

Suggestion to simplify. Organizations have names too. I added the recommendation here, in context, cf. `role`.

peterdesmet commented on content/docs/specifications/data-package.md:
```diff
@@ -268,13 +268,17 @@ The people or organizations who contributed to this Data Package. It `MUST` be a
   - Note on semantics: use of the "author" property does not imply that that person was the original creator of the data in the data package - merely that they created and/or maintain the data package. It is common for data packages to "package" up data from elsewhere. The original origin of the data can be indicated with the `sources` property - see above.
 - `organization`: a string describing the organization this contributor is affiliated to.
 
+A data producer is `RECOMMENDED` to follow [BibTeX](https://www.bibtex.com/f/author-field/) guidance when providing information about a person's name and title.
```
```suggestion
```

peterdesmet commented on content/docs/specifications/data-package.md:
```diff
@@ -268,13 +268,17 @@ The people or organizations who contributed to this Data Package. It `MUST` be a
 - `organization`: a string describing the organization this contributor is affiliated to.
 
+A data producer is `RECOMMENDED` to follow [BibTeX](https://www.bibtex.com/f/author-field/) guidance when providing information about a person's name and title.
+
```
```suggestion
```

peterdesmet reviewed (changes requested):
Review of text.

peterdesmet commented:
@roll Great! I've made some suggestions to the wording, to align with what's already there.

peterdesmet commented on content/docs/specifications/data-package.md:
````diff
@@ -268,7 +268,9 @@ The people or organizations who contributed to this Data Package. It `MUST` be a
 ```
 
-- `title`: name/title of the contributor (name for person, name/title of organization)
+- `title`: name of the contributor. It's `RECOMMENDED` to follow [BibTeX name format](https://www.bibtex.com/f/author-field/) when providing a person's name in `title`.
````
```suggestion
- `title`: name of the contributor. It's `RECOMMENDED` to follow [BibTeX name format](https://www.bibtex.com/f/author-field/) when providing a person's name.
```

peterdesmet replied:
One more correction 😄

roll commented:
@pschumm
@ezwelty
@pwalsh
@nichtich
Can you please take a look

ezwelty commented:
I have a few thoughts on this. As others have noted, `title` is the only universal way of representing a person's name. If a more granular model is needed for generating citations:

- I would NOT name the properties `firstName`, `lastName`.
- I would NOT suggest people muck up `title` to match some formatting convention.
- I would approve either one of:
  - Add two new properties `givenName`, `familyName`
  - Add a single `citationName` in Bibtext format
  - ... where in either case, use of `title` is encouraged to make explicit name order and secondary name parts.

My reasoning below.

Citation conventions require the family name (aka surname), which is not the *last* name in notable cases (examples below).
  - Most modern [Chinese names](https://en.wikipedia.org/wiki/Chinese_name) *begin* with a single-symbol family name followed by the given name. As a result of the last name assumption, Chinese authors are sometimes cited by their given name.
  - In most Spanish-speaking countries, most people have two family names, with the first often taking precedence. So for example Guillermo Cobos Campos should in Bibtex format be cited as "Cobos Campos, Guillermo" or "Cobos, Guillermo", but certainly not "Campos, Guillermo Cobos" (as is [frequently found](https://scholar.google.com/scholar?q=Guillermo+Cobos+Campos)).

At the [WGMS](https//wgms.ch), where we work with scientists from all over the world, we store their full name (equivalent to `title`), and the part of their name (it must be a subset of their full name) that should be treated as the family name when citation formats require a distinction. This allows us to support either order of given and family name, names without family names, etc with just two fields. A similar approach that would more robustly deal with suffixes, non-dropping-particles and dropping-particles would be to have a full name (`title`: 'John Smith Jr.') and a full citation name in Bibtex format (`citation`: 'Smith, John Jr.').

Note that even CSL doesn't have an attribute to specify given-family name order (some software tries to infer this using hacks, e.g.  based on the script of the name); the only way to achieve this is with their `literal`, aka our `title`.

pschumm commented:
FWIW, I find @ezwelty's argument(s) above pretty convincing, and now agree that either `givenName` and `familyName` or `citationName` is the way to go. I don't have a strong preference between these.

peterdesmet commented:
@ezwelty thanks for clarifying. I think the `citation` field is the best approach then: it is flexible and clearly communicates that the purpose is for citations. Some questions/suggestions

1. **Definition**: `citation`: Name of the contributor in [BibTex author format](https://www.bibtex.com/f/author-field/) for use in citations.
2. **Name**: Should this property be called `citation` (quite broad), `author` (maybe confusing with a role), `bibtex`, or `bibtexAuthor`?
3. **Use**: I'm guessing that not providing this property doesn't directly mean that the contributor should not be included in a citation?

## Examples

```json
[
  {
    "title": "John Smith Jr.",
    "citation": "Smith, Jr, John" <- Suffix in 2nd place
  },
  {
    "title": "Guillermo Cobos Campos",
    "citation": "Cobos Campos, G"
  },
  {
    "title": "Monitoring, understanding and forecasting global biomass flows of aerial migrants (GloBAM)",
    "citation": "{GloBAM project}" <- Wrapped in {}
  }
]
```

nichtich commented:
I oppose building on top of BibTeX instead of CSL and I oppose adding another field with already same semantics as `title` we already have. As [summarized above](https://github.com/frictionlessdata/datapackage/pull/20#issuecomment-1913485416) CSL can be stripped to one field for the full name (we already have `title`, so this should be fixed) and two fields for parts of the name.

PietrH commented:
If I'm the blocker I'll capitulate. 

I propose we reopen this discussion when we come across a user story that doesn't fit in the CSL derived definitions.

roll commented:
@peterdesmet @PietrH 
Thanks a lot for making it work! I've updated the PR 

@ezwelty @pschumm 
Please take a look if it's ok now

roll commented:
ACCEPTED by WG (6/9)

ezwelty commented:
@roll Looks good. My only additional suggestion would be to mention, when linking to the CSL docs, to follow their guidance on where to put suffixes and particles, since we don't include dedicated fields for those.
